webapp/src/main/java/com/box/l10n/mojito/service/cache/TieredCache.java

```diff
@@ -0,0 +1,109 @@
+package com.box.l10n.mojito.service.cache;
+
+import com.google.common.base.Preconditions;
+import org.springframework.cache.Cache;
+import org.springframework.cache.support.AbstractValueAdaptingCache;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.Callable;
+
+/**
+ * Enables a tiered caching setup where items are stored in both caches and retrieval
+ * is done preferentially from the tier 1 cache first and then on cache miss from tier 2.
+ *
+ * Usually expected to be used with an ephemeral in-process memory cache as tier 1 and a
+ * durable/persistent tier 2 cache.
+ *
+ * @author garion
+ */
+public class TieredCache extends AbstractValueAdaptingCache {
```

Comment:
Spring doesn't expose something similar? Is it only in the annotation that you can chain caches to use?

Reply:
Using a @Cacheable annotation where you can chain caches seems to be the only way Spring supports this as far as I can tell. Though it's not very transparent in that case as to how the system deals with cache misses and re-population in a chained scenario.